UofT-Frosh-Orientation/orientation-website
# 626 optimize profile page using dynamic imports

## **Contribution:**

### **Issue:**
Closes #626 

### **Accomplishments:**
- Split the code into two files: `PageProfileFrosh.jsx` and `PageProfileLeedur.jsx` where `Profile.jsx` checks if the user is a leader or frosh and loads the correct page accordingly.
- In each component, there were many if (leedur === true) checks. I was able to get rid of most of those since the pages are now separately loaded.
- There was a `ProfileHeader` component that would take in the boolean props: `leedur` and `editButton`. I put this component in both `PageProfileFrosh.jsx` and `PageProfileLeedur.jsx` files and removed the leedur prop from both since its obvious which header is for which user now. 
- Since the file names changed, I had to update some imports from two files: `ScuntHome.jsx` and `ProfileEdit.jsx`

### **Visuals:** 
Same old profile for Frosh (I only loaded frontend hence missing info): 
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/f5a9b31e-ea47-4a0d-b58e-8d9b8a6c6c36)

Same old profile for Leader (I only loaded frontend hence missing info): 
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/cfe27ac3-17bd-4f55-9f1d-326aba3673d4)

## **Details:**

### **Expected Behavior:**
- You should be able to just log in or sign up as Frosh/Leader and access all components as necessary.
- All components should work and show up on your profile page including header, edit button, schedule, scunt team/token, etc.

### **Testing Steps:**

With backend:
- Run docker
- Head to sign up or log in page (uncomment login page in `util/pages.jsx` if need be) with your Frosh account
- Access the Profile page and see if any component is missing
- Do the same steps with a Leader account as well

Frontend only:
- Go to `util/pages.jsx` and uncomment the `<AuthorizedPage></AuthorizedPage>` tags for the `Profile.jsx` page
- For Frosh account: just run the front end
- For Leader account: change to `const leader = true` on line 15
- Access the Profile page and see if any component is missing

### **Complications:**
- Couldn't figure out the dynamic importing since Suspense and lazy were giving me errors like this one
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/de5c93a6-01d7-4002-9b4d-4272c58eaa70)
- The files are still pretty long with ~700 lines of code. If needed, I can further split the components
- Since I was using just front end to check, I'm not sure if all components are properly shown such as scunt teams/QR codes
- I think the leader dropdown that was recently merged is different from my working copy, so let me know if I need to steal that one and work on editing it

### **Resources:**
- Tried this for debugging lazy loading but no luck yet: [https://stackoverflow.com/questions/58297907/suspense-is-not-loading-content-after-render-reactjs](url)
